### ISSUE TITLE
app.js: what to do about multiple HTML files in a project?

### ISSUE THREAD

**Issue by JonFerraiolo** [maqetta/maqetta#3377]
@aerwin 

Right now app.js is included in each HTML file that a user creates. If they put logic that is specific to a particular HTML file (e.g., file1.html) such as doing a dojo.byId() or a dijit.byId(), that logic will likely break when they try to run a different HTML file in the previewer (or when deployed).

Some options that have been mentioned:
- Add comments to the top of app.js warning users about this situation. 
- Post docs that show users how to deal with a shared app.js file across multiple HTML pages using the same file. (e.g., if dojo.byId() returns null, fail gracefully, or look at window.location)
- Add a control to the new HTML file dialog, or maybe a new preference, that allows the user to decide what to do about custom JavaScript files: (a) no custom JS files, (b) include app.js, (c) include `<basename>.js` with each `<basename>.html`. (Note that this latter approach will be complicated because we allow SaveAs into different folders.)

**Comment by billreed63** [comment]
I would vote for a combinations of the first two bullets, adding comments to app.js that give an example of select the code path based on the window.location

**Comment by aerwin** [comment]
I'd also suggest if we're in there changing comments and such that we require some other common things users are likely to use like:

```
require(["dojo/ready", "dojo/dom", "dojo/dom-style", "dijit/registry"], function(ready, dom, domStyle, registry){
```

**Comment by JonFerraiolo** [comment]
Yeah, the 3rd bullet doesn't seem feasible for M8 and goes somewhat against keep-it-simple-stupid

**Comment by JonFerraiolo** [comment]
Also, I agree with Tony about beefing up the standard list of requires. I'd include dojo/query and dojo/_base/connect, also.

**Comment by JonFerraiolo** [comment]
@jhpedemonte: For M8 migration, maybe we can look at app.js in old workspaces and see if the app.js file exactly matches what we shipped with M7, and if so, then replace with the new M8 version? How hard would that be?

**Comment by JonFerraiolo** [comment]
Pushing to M9. Too late to mess with this for M8.

**Comment by JonFerraiolo** [comment]
In working on sample code for a customer, I've concluded that we should indeed do bullet 3 above - in particular, we should have a preference (default off) that tells Maqetta to create a `<basename>.js` file for each new HTML page. The idea is that common functions would go into app.js, and page-specific logic would go into `<basename>.js`. 

Part of that enhancement would be that we would need to relocate the references to all JavaScript files, but I'm pretty sure we already do that in RebuildPage.js.

 Note that if Maqetta doesn't create `<basename>.js` automatically, users can always go into source mode and manually enter a SCRIPT tag for `<basename>.js`.

**Comment by JonFerraiolo** [comment]
In thinking about this some more, related to this issue, long-term I would think that whenever we create a CSS or JavaScript file, we should ask the user which HTML files should be modified to include a reference (LINK or SCRIPT tag) to the newly created file, and whenever a user renames a CSS or JavaScript file, we should ask the user whether to update the existing references within existing HTML files.

In other words, why can't Maqetta help the user out - it's a productivity aid, after all.
